src/navigator/types.tsx

```diff
@@ -272,7 +272,7 @@ export type StackParamList = {
       }
     | undefined
   [Screens.SwapScreenWithBack]: { fromTokenId: string } | undefined
-  [Screens.TabDiscover]: undefined
+  [Screens.TabDiscover]: { isTabNavigator?: boolean } | undefined
```

Comment:
Why does this have to be the union of the param that gets used and undefined? Do screens that use the same component need to have the same params? It seems like `isTabNavigator` is always used for the TabDiscover screen, and there are no props passed in ever for DAppsExplorerScreen, and then the component takes the union of these for `Props`

Reply:
That is a great question. I was following the pattern used in https://github.com/valora-inc/wallet/pull/5069, but I'll check with @satish-ravi as to why this was done.

Reply:
In this specific case since the options are `undefined` or an optional boolean parameter, it should be fine to have their own params. But if there are different params, there will be a type error when accessing params in the component. E.g., if props are `{a: string}` for one screen and `{b: string}` for another, the type of route.params will be `{a: string } | {b: string}` and TS won't let you do `route.params.a` or `route.params.b`. In any case this is only temporary while we switch to the tab navigator, we should generally not have screens sharing components